### DISCUSSION TITLE
How I can run with my portable Brave browser in python?

### DISCUSSION THREAD

**Discussion by FidoAnh** [seleniumbase/SeleniumBase#2224] [Q&A] [answered]
With Selenium I can set path to run which browser I want in python. How can I do that with Seleniumbase? And in undetected mode?

**Comment by mdmintz** [comment, marked as answer]
If you want to change the binary location, there's the `binary_location=PATH` arg:

```python
from seleniumbase import Driver

driver = Driver(binary_location=PATH)
```

The supported browsers are Chrome, Edge, Firefox, and Safari. UC Mode is only for Chrome.

However, if you want to use SeleniumBase APIs from [SeleniumBase/help_docs/method_summary.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/method_summary.md) while customizing how you launch the driver, you can use the format that lets you override how you launch your driver: [SeleniumBase/help_docs/syntax_formats.md#-9-overriding-the-driver-via-basecase](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#-9-overriding-the-driver-via-basecase)

**Reply by FidoAnh** [reply]
Thank you!

**Reply by FidoAnh** [reply]
I have try and get this error:
Warning: The Chromium binary specified is NOT valid!
Expecting "brave-portable.exe" to be found in ['chrome.exe', 'chromium.exe'] for the browser / OS!
(Will use default settings...)
then It use my chrome not use my brave browser

**Reply by FidoAnh** [reply]
I try overriding, they work!

**Reply by FidoAnh** [reply]
The problem is overriding use BaseCase so automatically called for opening and closing web browsers at the start and end of tests. I want something like Driver, I can call it when ever I want, the browser still there

**Reply by mdmintz** [reply]
You can still use UC Mode. Add `--uc` as a `pytest` command-line option for that.

**Reply by FidoAnh** [reply]
So how can I use it as Driver?

**Reply by mdmintz** [reply]
You have to use one of the supported browsers. SeleniumBase has the same supported browsers as Selenium:
Chrome, Edge, Firefox, Safari.

**Reply by FidoAnh** [reply]
but Brave is base on Chromium like chrome, and selenium support it, I can run it with selenium via chromedriver but I want to be undetected automation:(( I try look around today and find seleniumbase
